File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/ReattemptInstructionsMapper.cs
using ReattemptInstructionsSdk = OnlinePayments.Sdk.Domain.ReattemptInstructions;
using ReattemptInstructionsDto = Business.Domain.Payments.PaymentDetails.ReattemptInstructions;

namespace Infrastructure.Mappers.PaymentDetails;

public static class ReattemptInstructionsMapper
{
    public static ReattemptInstructionsDto Map(ReattemptInstructionsSdk? response)
    {
        return new()
        {
            Conditions = ReattemptInstructionsConditionsMapper.Map(response?.Conditions),
            FrozenPeriod = response?.FrozenPeriod,
            Indicator = response?.Indicator
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/ReattemptInstructionsMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\ReattemptInstructions as ReattemptInstructionsDto;
use OnlinePayments\Sdk\Domain\ReattemptInstructions as ReattemptInstructionsSdk;

final class ReattemptInstructionsMapper
{
    public static function mapFromSdkResponse(?ReattemptInstructionsSdk $response): ?ReattemptInstructionsDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new ReattemptInstructionsDto();
        $dto->conditions = ReattemptInstructionsConditionsMapper::mapFromSdkResponse($response?->conditions ?? null);
        $dto->frozenPeriod = $response?->frozenPeriod ?? null;
        $dto->indicator = $response?->indicator ?? null;

        return $dto;
    }
}
